### PR TITLE
Mgdapi 1355 Updating the Golang Version to 1.16.2

### PR DESCRIPTION
**Pull request by Boomatang**
# Description

Update the go version being used in the operator to 1.16.2

**Jira** https://issues.redhat.com/browse/MGDAPI-1355

## Verify 
* code builds locally
* all docker images are created

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -112,8 +112,8 @@ GOBIN=$(shell go env GOBIN)
 endif
 
 .PHONY: setup/moq
-setup/moq:
-	go install github.com/matryer/moq
+setup/moq: code/gen
+	GO111MODULE=off go get github.com/matryer/moq
```
Comment (matskiv) [inline Makefile:116]: I think we shouldn't be changing `go install` to `go get` 
AFAIK:
`go install` installs moq binary to make it available via CLI
`go get` installs moq as library dependency

Comment (matskiv) [reply Makefile:116]: Also, why is `code/gen` added above?

File: make/osde2e.mk [hunk 2]
```diff
@@ -15,4 +15,4 @@ image/osde2e/build/push: image/osde2e/build image/osde2e/push
 
 .PHONY: test/compile/osde2e
 test/compile/osde2e:
-	CGO_ENABLED=0 go test -v -c -o managed-api-test-harness.test ./test/osde2e -ginkgo.noColor
```
Comment (matskiv) [inline make/osde2e.mk:18]: This change seems unrelated to Go version bump,
highlighting in case it was unintentionally added to the commit.

File: Makefile (hunk 1 above)
Comment (Boomatang) [reply Makefile:116]: I am sure Oleg, but on Monday this week or last week you came up with that fix we where having around moq. :confused:

File: make/osde2e.mk (hunk 2 above)
Comment (Boomatang) [reply make/osde2e.mk:18]: On go 1.16, go test will not compile if extra flags are give to the `-c` flag. This was an intentionally commit.

Comment (matskiv) [reply make/osde2e.mk:18]: Okay, good to know.
Thank you for explanation 👍

File: Makefile (hunk 1 above)
Comment (matskiv) [reply Makefile:116]: Yes, that's true, but later I realized my mistake and I thought we discussed it.
Sorry for the confusion.
